### PR TITLE
factor out common logging initialisation routine

### PR DESCRIPTION
**Pull request by LizardWizzard**
This contains a lowest common denominator of pageserver and safekeeper log
initialisation routines. It uses daemonize flag to decide where to
stream log messages. In case daemonize is true log messages are
forwarded to file. Otherwise streaming to stdout is used. Usage of
stdout for log output is the default in docker side of things, so make
it easier to browse our logs via builtin docker commands.

### CODE REVIEW

**Review by hlinnaka — APPROVED** [review]

**Comment by LizardWizzard** [comment]
Rebased on main to pass the tests

**Review by funbringer — APPROVED** [review]

File: pageserver/src/branches.rs [hunk 1]
```diff
@@ -17,13 +17,14 @@ use std::{
 use zenith_utils::zid::{ZTenantId, ZTimelineId};
 
 use log::*;
+use zenith_utils::logging;
 use zenith_utils::lsn::Lsn;
 
-use crate::logger;
-use crate::restore_local_repo;
+// use crate::logger;
```
Comment (SomeoneToIgnore) [inline pageserver/src/branches.rs:23]: Guess this can be removed.
Just to collect all nits in one place: there's also a `pagees` typo a bit below.

**Review by SomeoneToIgnore — APPROVED** [review]
I like how cargo deps get cleaner.


On a bit unrelated topic: I'm thinking about the ways I'm able to run logs with `debug` level with IT python tests, one of the simple ways I see it is to use `RUST_LOG`.
Do you know if slog supports this? Had tried adjusting python scripts myself to verify, but could not make it work, so wondering now.

**Comment by LizardWizzard** [comment]
> I like how cargo deps get cleaner.

I like this too. Need to research workspace_hack things (maybe we can get rid of it?) so further cleanups do not affect compile times

> Do you know if slog supports this?
I see https://github.com/slog-rs/envlogger. It says it's a env_log port to slog. 

> but could not make it work

I think there is no builtin support, see envlogger reference above. But there is one more caveat, tests use zenith cli, which spawns pageserver process and cleans env before that, so there is no easy way to pass env variable from cli/tests to pageserver. Maybe we should change that (either do not clean them or whitelist some known ones)

**Pull request merged**